### PR TITLE
Profile feature

### PR DESCRIPTION
**Pull request by dannyr3**
Added Email, Bio, and ProfilePic properties to the User.java Class.  Added UpdateUserInfo Method in UserController to saveOrUpdate ProfilePics. Added to the H2 database User columns to store Email, Bio, and ProfilePic.  Added data entries in the appropriate columns in data.sql.

### CODE REVIEW

File: src/main/java/com/revaturelabs/ask/user/UserController.java [hunk 1]
```diff
@@ -82,6 +91,34 @@ public User updateUser(@RequestBody User user, @PathVariable int id) {
...
+        
+        User updatedUser = userService.findById(id);
+
+        key = userService.uploadProfilePicture(image, updatedUser.getUsername());
+        
+        updatedUser.setProfilePic(key);
+        userService.update(updatedUser);
+        
+      } catch (IOException e) {
+        throw new ResponseStatusException(HttpStatus.NO_CONTENT, "IOException on File.");
```
Comment (AdamDKing) [inline src/main/java/com/revaturelabs/ask/user/UserController.java]: NO_CONTENT is not an error response -- replace it with something appropriate please

File: src/main/java/com/revaturelabs/ask/user/UserController.java [hunk 2]
```diff
@@ -16,12 +19,18 @@
...
 import org.springframework.web.server.ResponseStatusException;
 import com.revaturelabs.ask.question.Question;
 import com.revaturelabs.ask.tag.TagService;
 
 @RestController
 @RequestMapping(path = "/users")
+/*
+ * @MultipartConfig(fileSizeThreshold = 1024 * 1024 * 1, // 1 MB maxFileSize = 1024 * 1024 * 10, //
+ * 10 MB maxRequestSize = 1024 * 1024 * 15, // 15 MB location = "/")
+ */
```
Comment (AdamDKing) [inline src/main/java/com/revaturelabs/ask/user/UserController.java]: Either uncomment this (if it does something) or delete it

File: src/main/java/com/revaturelabs/ask/user/UserService.java [hunk 3]
```diff
@@ -78,5 +80,19 @@
    * @return The updated user object
    */
   User updateTags(User user);
+  
+  /**
+   * "updateUserInfo" is used to update user info
+   * @param A user object
+   * @return The updated User with new information
+   */
+  User updateUserInfo(User user);
```
Comment (AdamDKing) [inline src/main/java/com/revaturelabs/ask/user/UserService.java]: Doesn't update(User u) already exist?

File: src/main/java/com/revaturelabs/ask/user/UserServiceImpl.java [hunk 4]
```diff
@@ -156,5 +168,66 @@ public User updateTags(User user) {
 
     return updatedUser;
   }
+
+  @Override
+  public User updateUserInfo(User user) {
+    // TODO Auto-generated method stub
+    return null;
+  }
```
Comment (AdamDKing) [inline src/main/java/com/revaturelabs/ask/user/UserServiceImpl.java]: why is this here?

File: src/main/java/com/revaturelabs/ask/user/UserServiceImpl.java [hunk 5]
```diff
@@ -156,5 +168,66 @@ public User updateTags(User user) {
...
+        .standard()
+        .withCredentials(new AWSStaticCredentialsProvider(credentials))
+        .withRegion(Regions.US_EAST_1)
+        .build();
+    
+    //Conversion block from multipartfile to image, then upload to s3
+    try {
+      byte[] bytes = image.getBytes();
+      if (bytes == null) {
+        throw new ImageConflictException("Invalid image");
```
Comment (AdamDKing) [inline src/main/java/com/revaturelabs/ask/user/UserServiceImpl.java]: I'm not sure ImageConflictException is reasonable here.. what's the actual problem that occurs if getBytes() returns null?  We want an Exception that describes that

**Review by AdamDKing — CHANGES_REQUESTED** [review]
See comments.  We'll discuss how to configure environment for CI/CD once requested changes are made/discussed

File: src/main/java/com/revaturelabs/ask/question/QuestionController.java [hunk 6]
```diff
@@ -164,6 +164,25 @@ public Question createOrUpdate(@RequestBody Question question, @PathVariable int
           "Error during update/creation of question", e);
     }
   }
+  
+  /**
+   * Gets questions that experts have responded to
+   * 
+   * @param expertId is the ID of the desired expert
+   */
+  @GetMapping("/{expertId}/response")
```
Comment (AdamDKing) [inline src/main/java/com/revaturelabs/ask/question/QuestionController.java]: This mapping means that hitting the endpoint "/questions/1/response" gets a list of questions produced by expert id 1.  That's not good.  users/1/answeredQuestions or similar would be better.  You could potentially use a query string param

Comment (AdamDKing) [reply src/main/java/com/revaturelabs/ask/question/QuestionController.java]: /questions?answeredBy=expertId maybe?

**Review by AdamDKing — DISMISSED** [review]
Looks good except for the endpoint, which should be changed

**Review by JordanAdkins — APPROVED** [review]

**Pull request merged**